### PR TITLE
Add xkcd comic retrieval

### PR DESCRIPTION
**Pull request by fractalpenguin**
**Features:**

- Retrieve latest xkcd comic using `:xkcd` 
latest

- Retrieving any xkcd comic available on xkcd.com by appending the comic number e.g `:xkcd 1618`
number

- Retrieving a random xkcd comic using `:xkcd ?`
random

Unlike the standard Discord xkcd embed, the bot provides the number (in case of the latest comic), the date of publication, and a link to the explainxkcd.com page for an explanation.

### CODE REVIEW

**Comment by AnonGuy** [comment]
You should probably be using `aiohttp` for this, rather than `requests`. `requests` is blocking, whereas [`aiohttp`](https://docs.aiohttp.org/en/docs/index.html) is async.

File: bot/cogs/fun.py [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
 
-from discord import Message
+from discord import Message, Embed
 from discord.ext.commands import (BadArgument, Bot, Context, EmojiConverter,
                                   command)
 
 from bot.constants import EVERYONE_REACTIONS
 
+from requests import get
+from random import randint
```
[the remark this answers is not in the document]
Comment (fractalpenguin) [reply bot/cogs/fun.py]: Done.

[the remark this answers is not in the document]
Comment (fractalpenguin) [reply bot/cogs/fun.py]: Done.

File: bot/cogs/fun.py [hunk 2]
```diff
@@ -17,7 +20,9 @@ def __init__(self, bot: Bot):
 
     async def on_message(self, message: Message):
         # React if a message contains an @here or @everyone mention.
-        if any(mention in message.content for mention in ("@here", "@everyone")):
+        if any(
+                mention in message.content
+                for mention in ("@here", "@everyone")):
```
[the remark this answers is not in the document]
Comment (fractalpenguin) [reply bot/cogs/fun.py]: Done, although there is no newline between the opening parenthesis and `mention`.

File: bot/cogs/fun.py [hunk 3]
```diff
@@ -26,9 +31,9 @@ def __init__(self, bot: Bot):
             await message.add_reaction("🤔")
 
     @command()
-    async def lmgtfy(self, ctx: Context, search_text: str, *args):
+    async def lmgtfy(search_text: str, *args):
         """
-        Lets the bot google that for you.
+        Lets the bot google tself, ctx: Context,hat for you.
```
[the remark this answers is not in the document]
Comment (fractalpenguin) [reply bot/cogs/fun.py]: Changed it.

**Pull request merged**